src/app/pcasts/dao/users_dao.py

```diff
@@ -68,16 +90,32 @@ def search_users(search_name, offset, max_search):
   return possible_users
 
 def change_user_name(user_id, new_name):
-    user = User.query.filter(User.id == user_id).first()
-    if user:
-      user.username = new_name
-      db_utils.db_session_commit()
-      return user
-    else:
-      raise Exception("The given user_id is Invalid")
+  user = User.query.filter(User.id == user_id).first()
+  if user:
+    user.username = new_name
+    db_utils.db_session_commit()
+    return user
+  else:
+    raise Exception("The given user_id is Invalid")
 
 def get_number_users():
-    return User.query.count()
+  return User.query.count()
 
 def get_all_users():
-    return User.query.filter().all()
+  return User.query.filter().all()
+
+def add_facebook_login(user, facebook_info):
+  if 'id' not in facebook_info:
```

Comment:
While this might be unnecessary right now - we might want to make a more sophisticated Facebook driver for interacting with the graph API that also does error handling like this on specific responses from the Facebook graph API.  We'll need it for other social features that @eileendai is designing for.